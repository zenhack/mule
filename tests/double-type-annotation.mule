# TODO: the discussion below is based on a different architecture for the type
# checker; reframe this in terms of how the type system works now.
#
# At one point we had a bug whose cause was that we'd flipped the arguments
# to a require_subtype. The upshot was that you could break the type system
# by using nested type annotations, where the outer was a subtype of the
# inner. This is a regression test; the following should be rejected:
(4 : int) : all a. a
