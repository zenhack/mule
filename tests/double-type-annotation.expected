
Desugared:

	((: (all a a)) ((: int) (Integer 4)))
Type error: Can't instantiate
